test.js

```diff
@@ -17,3 +17,61 @@ test('main', t => {
 	t.false(isAbsoluteUrl('C:\\Dev\\test-broken'));
 	t.false(isAbsoluteUrl('ht,tp://sindresorhus.com'));
 });
+
+test('HttpOnly set as true (same as default behavior). Assert TRUE tests', t => {
+	t.true(isAbsoluteUrl('http://sindresorhus.com', true));
+	t.true(isAbsoluteUrl('https://sindresorhus.com', true));
+	t.true(isAbsoluteUrl('httpS://sindresorhus.com', true));
+	t.true(isAbsoluteUrl('https://www.sindresorhus.com', true));
+	t.true(isAbsoluteUrl('https://www.sindres.orhus.com/path', true));
+	t.true(isAbsoluteUrl('httpS://www.sindresorhus.com', true));
+	t.true(isAbsoluteUrl('https://sindresorhus', true));	// Should this behavior exist? missing TLD?
+	t.true(isAbsoluteUrl('http://sin dresorhus.com', true));
+});
+
+test('HttpOnly set as true (same as default behavior). Assert FALSE tests', t => {
+	t.false(isAbsoluteUrl('yhttp://sindresorhus.com', true));
+	t.false(isAbsoluteUrl('file://sindresorhus.com', true));
+	t.false(isAbsoluteUrl('mailto:someone@example.com', true));
+	t.false(isAbsoluteUrl('data:text/plain;base64,SGVsbG8sIFdvcmxkIQ%3D%3D', true));
+	t.false(isAbsoluteUrl('ht,tp://sindresorhus.com', true));
+	t.false(isAbsoluteUrl('//sindresorhus.com', true));
+	t.false(isAbsoluteUrl('/foo/bar', true));
+	t.false(isAbsoluteUrl('foo/bar', true));
+	t.false(isAbsoluteUrl('foo', true));
+	t.false(isAbsoluteUrl('c:\\', true));
+	t.false(isAbsoluteUrl('c:\\Dev\\test-broken', true));
+	t.false(isAbsoluteUrl('C:\\Dev\\test-broken', true));
+	t.false(isAbsoluteUrl('h t,tp://sindresorhus.com', true));
+	// eslint-disable-line no-script-url
+	t.false(isAbsoluteUrl('javascript:throw%20document.cookie', true));
+});
+
+test('HttpOnly set as False. Assert TRUE tests', t => {
+	t.true(isAbsoluteUrl('http://sindresorhus.com', false));
+	t.true(isAbsoluteUrl('https://sindresorhus.com', false));
+	t.true(isAbsoluteUrl('httpS://sindresorhus.com', false));
+	t.true(isAbsoluteUrl('file://sindresorhus.com', false));
+	t.true(isAbsoluteUrl('mailto:someone@example.com', false));
+	t.true(isAbsoluteUrl('data:text/plain;base64,SGVsbG8sIFdvcmxkIQ%3D%3D', false));
+	t.true(isAbsoluteUrl('https://www.sindresorhus.com', false));
+	t.true(isAbsoluteUrl('https://www.sindres.orhus.com/path', false));
+	t.true(isAbsoluteUrl('httpS://www.sindresorhus.com', false));
+	// eslint-disable-line no-script-url
+	t.true(isAbsoluteUrl('javascript:throw%20document.cookie', false));
+
+	t.true(isAbsoluteUrl('https://sindresorhus', false));	// Should this behavior exist? missing TLD?...
```

Comment:
Instead of asking questions, try to research and figure it out.

Reply:
I felt it should at least check for the TLD, but it would return false to URLs as `http://localhost:7700` and they are absolute. So it will not check for it.